feat(module-tools): support to use dts respectExternal (#3876)
web-infra-dev/modern.js: packages/document/module-doc/docs/en/api/config/build-config.md

- Type: `boolean`
- Default: `false`

### respectExternal

> When set to `false`, three-party package types will be excluded from bundle, when set to `true`, it will determine whether three-party types need to be bundled based on[externals](#externals).

Reviewer: ```suggestion
When set to `false`, the type of third-party packages will be excluded from the bundle, when set to `true`, it will determine whether third party types need to be bundled based on [externals](#externals).
```

When bundle d.ts, export is not analyzed, so any three-way package type you use may break your build, which is obviously uncontrollable.
So we can avoid it with this configuration.

> - Type: `boolean`

Reviewer: ```suggestion
- **Type**: `boolean`
```

> - Default value: `true`

Reviewer: ```suggestion
- **Default**: `true`
```

With `inject` turned on, you need to pay attention to the following points.

- `@import` in the css file will not be processed. So if you have `@import` in your css file, then you need to bring in the css file manually in js (less,scss files don't need it because they will be preprocessed).

> - The impact of `sideEffects` needs to be considered, by default our builder will consider it as a side effect, if the `sideEffects` field set in your project or in the package.json of the three-party package and it does not contain this css file, then you will get a warning

Reviewer: ```suggestion
- The impact of `sideEffects` needs to be considered, by default our builder will consider it as a side effect, if the `sideEffects` field set in your project or in the package.json of the third party package and it does not contain this css file, then you will get a warning
```

Author: so,  third party or three-party?

Author: Fine, i' ll replace all three-party with third-party

Reviewer: Yes, the correct term is "third party". "three party" is not a commonly used term in English.
